R/utilities.R

```diff
@@ -0,0 +1,346 @@
+    
+    if (is.null(new_fac_spellings)) {
+        new_fac_spellings <- read_new_fac_spellings()
+    }
+    if (is.null(old_fac_info)) {
+        old_fac_info <- behindbarstools::read_fac_info()
+    }
+    if (is.null(old_fac_spellings)) {
+        old_fac_spellings <- behindbarstools::read_fac_spellings()
+    }
+    
+    out <- new_fac_spellings %>% 
+        rowwise() %>% 
+        mutate(exists_ = behindbarstools::is_fac_name(xwalk_name_clean, State, old_fac_info, old_fac_spellings, include_alt = FALSE),
+               valid_state_ = behindbarstools::is_valid_state(State), 
+               warning_msg_ = paste0(xwalk_name_clean, " (", State, ")"))
+    
+    drop <- out %>% 
+        filter(exists_ == FALSE | valid_state_ == FALSE)
+    
+    if (length(nrow(drop)) > 0) {
+        for (i in 1:nrow(drop)) {
+            if (drop[i,]$exists_ == FALSE) {
+                warning(stringr::str_c(
+                    "Dropping because clean facility name does not exist in fac_info: ", drop[i,]$warning_msg_))
+            }
+            if (drop[i,]$valid_state_ == FALSE) {
+                warning(stringr::str_c(
+                    "Dropping because state name is invalid: ", drop[i,]$warning_msg_)) 
+            }
+        }
+    }
+    
+    out %>% 
+        filter(exists_ == TRUE & valid_state_ == TRUE) %>% 
+        select(
+            -exists_, 
+            -valid_state_, 
+            -warning_msg_
+        )
+}
+
+generate_new_fac_id <- function(old_fac_info = NULL, old_fac_spellings = NULL) {
+    #' Generate next facility ID 
+    #' 
+    #' Generates the next facility ID by adding one to the largest existing ID. 
+    #'
+    #' @param old_fac_info data frame, existing facility info  
+    #' @param old_fac_spellings data frame, existing facility spellings 
+    #'
+    #' @return
+    
+    if (is.null(old_fac_info)) {
+        old_fac_info <- behindbarstools::read_fac_info()
+    }
+    if (is.null(old_fac_spellings)) {
+        old_fac_spellings <- behindbarstools::read_fac_spellings()
+    }
+    
+    max_id <- max(old_fac_info$Facility.ID, old_fac_spellings$Facility.ID, na.rm = TRUE)
+    return (max_id + 1) 
+}
+
+populate_new_fac_info <- function(new_fac_info = NULL, hifld_data = NULL) {
+    #' Populate new facility data 
+    #' 
+    #' Populates missing data in the new facility data updates from the HIFLD 
+    #' database and by geocoding addresses. Assigns a new facility ID to each entry. 
+    #'
+    #' @param new_fac_info data frame, new facility info updates
+    #' @param hifld_data data frame with HIFLD data 
+    #'
+    #' @return data frame with missing data populated 
+    
+    if (is.null(new_fac_info)) {
+        new_fac_info <- read_new_fac_info()
+    }
+    if (is.null(hifld_data)) {
+        hifld_data <- behindbarstools::read_hifld_data()
+    }
+    
+    new_fac_info %>%
+        # Populate from HIFLD 
+        rowwise() %>% 
+        mutate(
+            Population.Feb = coalesce(
```

Comment:
Going back to our earlier today conversation in the data team meeting, think we should add in `population_source` indicator here